RLBotFramework/parsing/agent_config_parser.py

```diff
@@ -0,0 +1,176 @@
+                                             possessed_human - controlled by the framework but the game detects it as a human""")
+
+    participant_header.add_value(PARTICIPANT_BOT_SKILL_KEY, float, default=1.0,
+                                 description='If participant is a bot and not RLBot controlled,' +
+                                             ' this value will be used to set bot skill.\n' +
+                                             '0.0 is Rookie, 0.5 is pro, 1.0 is all-star. ' +
+                                             ' You can set values in-between as well.')
+
+
+def get_sanitized_bot_name(dict, name):
+    """
+    Cut off at 31 characters and handle duplicates.
+    :param dict: Holds the list of names for duplicates
+    :param name: The name that is being sanitized
+    :return: A sanitized version of the name
+    """
+    if name not in dict:
+        new_name = name[:31]  # Make sure name does not exceed 31 characters
+        dict[name] = 1
+    else:
+        count = dict[name]
+        new_name = name[:27] + "(" + str(count + 1) + ")"  # Truncate at 27 because we can have up to '(10)' appended
+        dict[name] = count + 1
+
+    return new_name
+
+
+def get_team(config, index):
+    """
+    Returns which team the bot is on (represented by an integer)
+    """
+    return config.getint(PARTICIPANT_CONFIGURATION_HEADER, PARTICIPANT_TEAM, index)
+
+
+def get_bot_config_bundle(bot_config_path):
+    raw_bot_config = configparser.RawConfigParser()
+    raw_bot_config.read(bot_config_path)
+    config_directory = os.path.dirname(os.path.realpath(bot_config_path))
+    return BotConfigBundle(config_directory, raw_bot_config)
+
+
+def get_bot_config_bundles(num_participants, config, config_bundle_overrides):
+    """
+    Adds all the config files or config objects.
+    :param num_participants:
+    :param config:
+    :param config_bundle_overrides: These are configs that have been loaded from the gui, they get assigned a bot index.
+    :return:
+    """
+    config_bundles = []
+    for i in range(num_participants):
+        if i in config_bundle_overrides:
+            config_bundles.append(config_bundle_overrides[i])
+            logger.debug("Config available")
+        else:
+            bot_config_path = config.get(PARTICIPANT_CONFIGURATION_HEADER, PARTICIPANT_CONFIG_KEY, i)
+            config_bundles.append(get_bot_config_bundle(bot_config_path))
+            logger.debug("Reading raw config")
+
+    return config_bundles
+
+
+def get_bot_options(bot_type):
+    if bot_type == 'human':
+        is_bot = False
+        is_rlbot = False
+    elif bot_type == 'rlbot':
+        is_bot = True
+        is_rlbot = True
+    elif bot_type == 'psyonix':
+        is_bot = True
+        is_rlbot = False
+    elif bot_type == 'possessed_human':
+        # this is an rlbot but a very specific rlbot
+        is_bot = True

```

Comment:
is_bot = False

Reply:
is_bot will still be true.
Because it is running an agent (notice the comment right above that line :P)

Reply:
I feel that training_bot is a bad name, and also that possessed_human is a bad name for what you're using it for.

I originally came up with "possessed_human" as a short way of saying "a 'human' player from the game's perspective which is actually controlled by rlbot". Obviously possessed_human was not an ideal name because it got lost in translation when you wrote this code. I'd say it's an even worse name for "a bot that drives based on input from a controller".

I'm not a fan of training_bot because it could (rarely) be used outside of training mode, and generally I don't like mixing up what something is with how it's expected to be used.